scripts/demangle.py

```diff
@@ -10,62 +12,56 @@
 from source_code import *
 
 
-# get source code from docker
-def get_source_from_docker(
-    file_path: str, file_name: str, proj_name: str
-) -> Optional[str]:
-    # Create a container from the image
+# Get all files from docker
+def copy_files_from_docker(proj_name: str, output_path: str) -> bool:
     client = docker.from_env()
     image = client.images.get(f"gcr.io/oss-fuzz/{proj_name}:latest")
     container = client.containers.run(image, detach=True)
-    # check file_path, it may not the absolute path
-    if "/" not in file_path or file_path[0] != "/":
-        # Hard coding possible file path to retrive file
-        file_path4 = "/src"
-        file_path3 = os.path.join("/src", file_path)
-        file_path2 = os.path.join("/src", proj_name, "src", file_path)
-        file_path = os.path.join("/src", proj_name, file_path)
+    try:
+        stream, _ = container.get_archive("/src")
+    except docker.errors.NotFound:
+        error(f"Docker  {proj_name} path /src  NOT found\n")
+        return False
 
-        build_sh = ""
-        for chunk in container.get_archive("/src/build.sh")[0]:
-            build_sh += chunk.decode("utf-8")
-        pattern = r"cd\s+(\S+)"
-        matches = re.findall(pattern, build_sh)
-        for match in matches:
-            file_path4 = os.path.join(file_path4, match)
+    fileobj = io.BytesIO()
+    for chunk in stream:
+        fileobj.write(chunk)
+    fileobj.seek(0)
 
-    # Try those possible path
-    try:
-        code_content = ""
+    # Extract files from the file-like object
+    with tarfile.open(fileobj=fileobj) as tar:
+        tar.extractall(output_path)
 
-        for chunk in container.get_archive(file_path)[0]:
-            code_content += chunk.decode("utf-8")
-    except docker.errors.NotFound:
-        try:
-            code_content = ""
-            for chunk in container.get_archive(file_path2)[0]:
-                code_content += chunk.decode("utf-8")
-        except docker.errors.NotFound:
-            try:
-                code_content = ""
-                for chunk in container.get_archive(file_path3)[0]:
-                    code_content += chunk.decode("utf-8")
-            except docker.errors.NotFound:
-                try:
-                    code_content = ""
-                    for chunk in container.get_archive(file_path4)[0]:
-                        code_content += chunk.decode("utf-8")
-                except docker.errors.NotFound:
-                    code_content = None
-                    error(f"File  {file_path}  NOT found\n")
-    # Stop and remove the container
     container.stop()
     container.remove()
-    return code_content
+    return True
+
+
+# get source code from docker
```

Comment:
this function is getting path from `OSSFUZZ_SCRIPTS/output/proj_name/codes`, not docker right?

Reply:
You mean this function `get_source_code_path`? If so, yes, it gets the path from local.